impress-org/givewp
# Feature: add recurring donation options migration step

## Description



This adds a recurring donation options migration step to convert v2 form meta to v3 form block attributes.

## Affects



The amount block (recurring donation settings) during form migration.

## Visuals



**v2 Form**

<img width="968" alt="Screenshot 2023-08-30 at 3 32 11 PM" src="https://github.com/impress-org/givewp/assets/10138447/e96d5899-7dde-4e05-ae3b-fa90f8d08b92">

<img width="798" alt="Screenshot 2023-08-30 at 3 33 14 PM" src="https://github.com/impress-org/givewp/assets/10138447/a67a1abf-365a-4415-9522-642c95e107ce">

**v3 Form Migrated**

<img width="1113" alt="Screenshot 2023-08-30 at 3 32 28 PM" src="https://github.com/impress-org/givewp/assets/10138447/081c26f7-171e-4a47-a444-88beec287ed0">

<img width="747" alt="Screenshot 2023-08-30 at 3 33 22 PM" src="https://github.com/impress-org/givewp/assets/10138447/9ec1c7b5-3c1a-4f00-b42d-b0d1ee8d0ce4">


## Testing Instructions



1. Create a v2 form w/ give-recurring installed/enabled
2. Modify the recurring options in small increments followed by running `wp givewp form:migrate {formId}`
3. Check the v3 form to make sure the migrated settings make sense
4. Repeat steps 2-3 until all settings have been tested & compared.

## Pre-review Checklist






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205353925025788
  - https://app.asana.com/0/0/1205396809084390
